subsys/suit/storage/src/suit_storage_nrf9280.c

```diff
@@ -0,0 +1,1085 @@
+			suit_role_name_get(role));
+		return err;
+	}
+
+	err = suit_storage_envelope_install(area_addr, area_size, id, addr, size);
+	if (err != SUIT_PLAT_SUCCESS) {
+		LOG_ERR("Failed to install envelope with role 0x%x%s.", role,
+			suit_role_name_get(role));
+		return err;
+	}
+
+	LOG_INF("Envelope with role 0x%x%s saved.", role, suit_role_name_get(role));
+
+	return err;
+}
+
+suit_plat_err_t suit_storage_var_get(size_t index, uint32_t *value)
+{
+	struct suit_storage_app *app_storage = (struct suit_storage_app *)SUIT_STORAGE_APP_ADDRESS;
+	size_t area_size = offsetof(struct suit_storage_nvv, digest);
+	uint8_t *area_addr = (uint8_t *)digest_struct_find(
+		(uint8_t *)&app_storage->app.nvv, (uint8_t *)&app_storage->app.nvv_bak,
+		offsetof(struct suit_storage_nvv, digest));
+
+	return suit_storage_nvv_get(area_addr, area_size, index, value);
+}
+
+suit_plat_err_t suit_storage_var_set(size_t index, uint32_t value)
+{
+	struct suit_storage_app *app_storage = (struct suit_storage_app *)SUIT_STORAGE_APP_ADDRESS;
+	size_t area_size = offsetof(struct suit_storage_nvv, digest);
+	uint8_t *backup_addr = (uint8_t *)&app_storage->app.nvv_bak;
+	uint8_t *area_addr =
+		(uint8_t *)digest_struct_find((uint8_t *)&app_storage->app.nvv, backup_addr,
+					      offsetof(struct suit_storage_nvv, digest));
+
+	/* It is not allowed to update NVVs if the area has incorrect digest. */
+	if (area_addr == backup_addr) {
+		return SUIT_PLAT_ERR_IO;
+	}
+
+	suit_plat_err_t err = suit_storage_nvv_set(area_addr, area_size, index, value);
+
+	if (err != SUIT_PLAT_SUCCESS) {
+		LOG_ERR("Failed to update NVV at index %d", index);
+		return err;
+	}
+
+	/* Update digest and backup. */
+	return digest_struct_commit(area_addr, backup_addr, area_size);
+}
+
+suit_plat_err_t suit_storage_report_clear(size_t index)
+{
+	uint8_t *area_addr = NULL;
+	size_t area_size = 0;
+	suit_plat_err_t err;
+
+	err = find_report_area(index, &area_addr, &area_size);
+	if (err != SUIT_PLAT_SUCCESS) {
+		LOG_WRN("Unable to find report area at index %d.", index);
+		return err;
+	}
+
+	return suit_storage_report_internal_clear(area_addr, area_size);
+}
+
+suit_plat_err_t suit_storage_report_save(size_t index, const uint8_t *buf, size_t len)
+{
+	uint8_t *area_addr = NULL;
+	size_t area_size = 0;
+	suit_plat_err_t err;
+
+	err = find_report_area(index, &area_addr, &area_size);
+	if (err != SUIT_PLAT_SUCCESS) {
+		LOG_WRN("Unable to find area for report at index %d.", index);
+		return err;
+	}
+
+	return suit_storage_report_internal_save(area_addr, area_size, buf, len);
+}
+
+suit_plat_err_t suit_storage_report_read(size_t index, const uint8_t **buf, size_t *len)
+{
+	uint8_t *area_addr = NULL;
+	size_t area_size = 0;
+	suit_plat_err_t err;
+
+	err = find_report_area(index, &area_addr, &area_size);
+	if (err != SUIT_PLAT_SUCCESS) {
+		LOG_WRN("Unable to find area with report at index %d.", index);
+		return err;
+	}
+
+	return suit_storage_report_internal_read(area_addr, area_size, buf, len);
+}
```

Comment:
Note that in the most recent version, a new API (`suit_plat_err_t suit_storage_purge(suit_manifest_domain_t domain)`) emerged.